Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -452,7 +452,11 @@ void TextEditor::paint_event(PaintEvent& event)
         painter.draw_line(ruler_rect.top_right(), ruler_rect.bottom_right(), palette().ruler_border());
     }
 
-    painter.translate(-horizontal_scrollbar().value(), -vertical_scrollbar().value());
+    auto horizontal_scrollbar_value = horizontal_scrollbar().value();
+    painter.translate(-horizontal_scrollbar_value, -vertical_scrollbar().value());
+    if (m_icon && horizontal_scrollbar_value > 0) {
+        painter.translate(min(icon_size() + icon_padding(), horizontal_scrollbar_value), 0);
+    }
```

Comment:
Please omit curly braces for single line bodies

Reply:
Resolved. Hopefully I eventually learn how to properly name commits.

Reply:
> Hopefully I eventually learn how to properly name commits.

It's hard, and I'm picky :^)
Made some small changes to replace favicon with icon, favicon is a Browser-specific concept.